### DISCUSSION TITLE
Is it possible for both Jest & @cypress/react unit tests to coexist in a typescript project?

### DISCUSSION THREAD

**Discussion by glomotion** [cypress-io/cypress#15063] [Component Testing] [answered]
I spent an entire day yesterday trying to do just that. I just couldn't find a way to prevent the cypress types from conflicting with the jest types. (global expect etc).

I tried splitting the @cypress/react unit tests out and into a completely seperate working directory with it's own tsconfig, but then cypress error'd for me saying that the component tests **must** be in the same place as the components. 

If i'm being honest, I truly love using @cypress/react and I was hoping to transition our developers into using it over jest/js-dom - but the above problem has pretty much stopped me in my tracks. 

Has anyone else managed to pull this off?

**Comment by bahmutov** [comment, marked as answer]
Read https://glebbahmutov.com/blog/local-cypress/ which explains the global variables problem and what we are doing about it

**Reply by glomotion** [reply]
Omg that is amazing!!! Thank you!!!

**Reply by glomotion** [reply]
@bahmutov out of interest, are you aware of anything changing RE cypress & chai type definitions inside cypress 7.x?

Installing 7.x has once again conflicted all our jest types with chai & mocha types. :(
